core/domain/story_domain.py

```diff
@@ -362,7 +418,8 @@ def validate(self):
                 raise utils.ValidationError(
                     'Expected exploration ID to be a string, received %s' %
                     self.exploration_id)
-        self.require_valid_thumbnail_filename(self.thumbnail_filename)
+        if self.thumbnail_filename is not None:
+            self.require_valid_thumbnail_filename(self.thumbnail_filename)
```

Comment:
The type of `self.thumbnail_filename` is `Optional[str]`, but `require_valid_thumbnail_filename()` can only accept string values. So, to narrow down the type i used `if` clause here. Also, it is expected from `self.thumbnail_filename` to hold None values, so i think we don't have to check it against None value. 

a similar approach is used in topic_domain:
https://github.com/oppia/oppia/blob/2af5fb246c16ad2f0a944c6577f44898e9109527/core/domain/topic_domain.py#L471